universum/github_handler.py

```diff
@@ -0,0 +1,61 @@
+import json
+import sys
+import requests
+
+from .modules.automation_server.jenkins_server import JenkinsServerForTrigger
+from .modules.vcs.github_vcs import GithubToken
+from .modules.output import needs_output
+from .modules.structure_handler import needs_structure
+from .lib.utils import make_block
+
+
+@needs_structure
+@needs_output
+class GithubHandler(JenkinsServerForTrigger, GithubToken):
+    """
+    Excepts payload from stdin!
+    """
+
+    @staticmethod
+    def define_arguments(argument_parser):
+        argument_parser.add_argument('--event', '-e', dest='event', metavar="GITHUB_EVENT",
+                                     help='Currently parsed from "x-github-event" header')
+        argument_parser.add_argument('--payload', '-p', dest='payload', metavar="PAYLOAD",
+                                     help='<actual help coming later> leave "-" to read from stdin')
+
+    def __init__(self, *args, **kwargs):
+        super(GithubHandler, self).__init__(*args, **kwargs)
+        if self.settings.payload == '-':
+            self.payload = json.loads(sys.stdin.read())
+        else:
+            self.payload = json.loads(self.settings.payload)
+
+    @make_block("Analysing trigger payload")
+    def execute(self):
+        if self.settings.event == "check_suite" and (self.payload["action"] in ["requested", "rerequested"]):
+            url = self.payload["repository"]["url"] + "/check-runs"
+            data = {"name": "CI tests", "head_sha": self.payload["check_suite"]["head_sha"]}
+            headers = {'Authorization': f"token {self.get_token(self.payload['installation']['id'])}",
+                       'Accept': 'application/vnd.github.antiope-preview+json'}
+            response = requests.post(url=url, json=data, headers=headers)
```

Comment:
Isn't it supposed to be easier to do this stuff using pygithub rather than POST-requests?

Reply:
Good question, I was sure I've researched it, but it doesn't seem right. I'll double check.

Reply:
Checked again. We're using this implementation:
https://github.com/PyGithub/PyGithub/blob/925764add54f557bfa446525a76ccb86180b577a/github/MainClass.py#L783
that doesn't provide any more functionality than already used. So, I guess, no? There's no easier way?

Reply:
Why don't we use another class then?

Reply:
Because we don't know of any other applicable classes.